osu.Game/Screens/Edit/Compose/Components/SelectionScaleHandler.cs

```diff
@@ -32,8 +32,26 @@ public partial class SelectionScaleHandler : Component
         /// </remarks>
         public Bindable<bool> CanScaleDiagonally { get; private set; } = new BindableBool();
 
+        /// <summary>
+        /// Whether scaling anchored by the selection origin can currently be performed.
+        /// </summary>
+        public Bindable<bool> CanScaleSelectionOrigin { get; private set; } = new BindableBool();
+
+        /// <summary>
+        /// Whether scaling anchored by the center of the playfield can currently be performed.
+        /// </summary>
+        public Bindable<bool> CanScalePlayfieldOrigin { get; private set; } = new BindableBool();
```

Comment:
- These are named kinda weird (as if the "origin" was being scaled).
- `CanScaleSelectionOrigin` is weird here to me because `CanScale{X,Y,Diagonally}` already exist and my presumption would be that those _already_ are for scaling about the origin. I'm not really even sure what it achieves, the only usage (`PreciseScalePopover`) could just inline the `CanScaleX.Value || CanScaleY.Value` logic locally.
- `CanScalePlayfieldOrigin` cannot live in the base `SelectionScaleHandler` imo because that only makes sense in the specific contexts where a playfield is present. Should be moved to the derived `OsuSelectionScaleHandler` imo.

The "not being able to access these if these are moved to derived" issue can be fixed by casting. I don't see that as a huge concern compared to just putting stuff in the base impl that makes no sense for skinning, for instance.